### PR TITLE
Added support to use model output dictionary keys from file

### PR DESCRIPTION
**Pull request by aozo**
### Background

Pylot currently assumes that the output from the pretrained TF models is a dictionary that contains the following keys: `boxes`, `scores`, `classes`,  and `detections`.

However, if we use a different model from the TF object detection zoo, the keys in the output dictionary is slightly different: `detection_boxes`, `detection_scores`, `detection_classes`, and `num_detections`.

Instead of modifying the `detection_operator` implementation to use the correct keys depending on the model we're using, we instead add a flag for the path of a file that specifies these keys.

### Summary of Changes
* Added a new flag called `path_model_outputs`. This points to a file that specifies the Python dictionary keys that Pylot should use to obtain the detection model outputs such as the bounding boxes, detection scores, class IDs of objects detected, and the number of detections.
* Added new files for the dictionary keys:
    * `pylot.outputs` are the defaults
    * `custom.outputs` can be defined by us depending on the detection model we are using.

### Validation
* Was able to successfully enable object detection using the [SSD ResNet50 V1 FPN 640x640 (RetinaNet50)](http://download.tensorflow.org/models/object_detection/tf2/20200711/ssd_resnet50_v1_fpn_640x640_coco17_tpu-8.tar.gz) model from the TF model zoo with the following `e2e.conf` flags:

```
--obstacle_detection_model_names=ssd_resnet50
--obstacle_detection_model_paths=dependencies/models/obstacle_detection/ssd_resnet50/
--path_coco_labels=dependencies/models/coco.names
--path_model_outputs=dependencies/models/custom.outputs
```
image

### CODE REVIEW

File: configs/e2e.conf [hunk 1]
```diff
@@ -3,6 +3,8 @@
 --nosimulator_obstacle_detection
 --obstacle_detection_model_paths=dependencies/models/obstacle_detection/faster-rcnn/
 --obstacle_detection_model_names=faster-rcnn
+--path_coco_labels=dependencies/models/pylot.names
```
Comment (aozo) [inline configs/e2e.conf:6]: This flag already existed, but added it here explicitly as a reminder that we may need to change this depending on the model that we are using & the class labels that are supported.

File: configs/e2e.conf [hunk 2]
```diff
@@ -3,6 +3,8 @@
...
+--path_model_outputs=dependencies/models/pylot.outputs
```
Comment (aozo) [inline configs/e2e.conf:7]: This is a new flag.

File: dependencies/models/custom.outputs [hunk 3]
```diff
@@ -0,0 +1,4 @@
+detection_boxes
+detection_scores
+detection_classes
+num_detections
```
Comment (aozo) [inline dependencies/models/custom.outputs:4]: These are the outputs from the SSD ResNet detection model. We can change this as needed depending on the model we end up using.

File: dependencies/models/pylot.outputs [hunk 4]
```diff
@@ -0,0 +1,4 @@
+boxes
+scores
+classes
+detections
```
Comment (aozo) [inline dependencies/models/pylot.outputs:4]: These are the default outputs from the existing Pylot detection models.

File: pylot/perception/detection/detection_operator.py [hunk 5]
```diff
@@ -206,10 +207,10 @@ def __run_model(self, image_np):
...
         result = infer(tf.convert_to_tensor(value=image_np_expanded))
 
-        boxes = result['boxes']
-        scores = result['scores']
-        classes = result['classes']
-        num_detections = result['detections']
+        boxes = result[self._model_outputs['boxes']]
+        scores = result[self._model_outputs['scores']]
+        classes = result[self._model_outputs['classes']]
+        num_detections = result[self._model_outputs['detections']]
```
Comment (aozo) [inline pylot/perception/detection/detection_operator.py:213]: Instead of using hard-coded keys into the `result` dictionary, we use keys from the `*.outputs` file specified by the new config flag.

**Comment by abrahamchan** [comment]
LGTM. I believe you have tested this on multiple models now.
